src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java

```diff
@@ -0,0 +1,103 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DOCTOR;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIMESLOT_START;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.FindAppointmentCommand;
+import seedu.address.logic.commands.FindPatientCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.AppointmentContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindAppointmentCommand object
+ */
+public class FindAppointmentCommandParser implements Parser<FindAppointmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindAppointmentCommand
+     * and returns a FindAppointmentCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindAppointmentCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindPatientCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DOCTOR,
+                PREFIX_TIMESLOT_START, PREFIX_TAG);
+
+        if (!areAnyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DOCTOR,
+                PREFIX_TIMESLOT_START, PREFIX_TAG)) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindAppointmentCommand.MESSAGE_USAGE));
+        }
+
+        ArrayList<String> patientKeywords = new ArrayList<String>();
+        ArrayList<String> doctorKeywords = new ArrayList<String>();
+        ArrayList<String> timeStartKeywords = new ArrayList<String>();
+        ArrayList<String> tagKeywords = new ArrayList<String>();
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String[] xs = listKeywords(argMultimap, PREFIX_NAME);
+            Collections.addAll(patientKeywords, xs);
+        }
+
+        if (argMultimap.getValue(PREFIX_DOCTOR).isPresent()) {
+            String[] xs = listKeywords(argMultimap, PREFIX_DOCTOR);
+            Collections.addAll(doctorKeywords, xs);
+        }
+
+        if (argMultimap.getValue(PREFIX_TIMESLOT_START).isPresent()) {
+            String[] xs = listKeywords(argMultimap, PREFIX_TIMESLOT_START);
+            Collections.addAll(timeStartKeywords, xs);
+        }
+
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            String[] xs = listKeywords(argMultimap, PREFIX_TAG);
+            Collections.addAll(tagKeywords, xs);
+        }
```

Comment:
this may be just me but i feel like doing the object creation on another line feels v messy...
i'd rather do `Collections.addAll(tagKeywords, listKeywords(argMultimap, PREFIX_XX));`, but its up to u!!